drivers/sensor/tmag5273/tmag5273.h

```diff
@@ -0,0 +1,452 @@
+#define TMAG5273_DIAG_NO_FAIL			0
+#define TMAG5273_DIAG_FAIL			1
+#define TMAG5273_CONVERSION_NOT_COMPLETE	0
+#define TMAG5273_CONVERSION_COMPLETE		1
+
+#define TMAG5273_INTB_PIN_LOW			0
+#define TMAG5273_INTB_PIN_HIGH			1
+#define TMAG5273_OSC_ERR			1
+#define TMAG5273_INT_ERR			1
+#define TMAG5273_OTP_CRC_ERR			1
+#define TMAG5273_VCC_UV_ERR			1
+
+/* additional values */
+#define TMAG5273_MEAS_RANGE_LOW_MT_VER1		40
+#define TMAG5273_MEAS_RANGE_HIGH_MT_VER1	80
+#define TMAG5273_MEAS_RANGE_LOW_MT_VER2		133
+#define TMAG5273_MEAS_RANGE_HIGH_MT_VER2	266
+
+#define TMAG5273_TEMPERATURE_T_SENS_T0		25
+#define TMAG5273_TEMPERATURE_T_ADC_T0		17508
+#define TMAG5273_TEMPERATURE_T_ADC_RES		60.1
+
+#define TMAG5273_T_START_SLEEP_US		50
+#define TMAG5273_T_GO_SLEEP_US			20
+
+/** OR this bit to any register address to trigger a conversion in standby mode */
+#define TMAG5273_CONVERSION_START_BIT		0x80
+
+/* -------------------------------------------------------------------- */
+
+/* --- helper macros --- */
+
+/**
+ * @brief   Macro to set bits in a register \a reg_data
+ *
+ * @param[in]  reg_data \c uint8_t register value
+ * @param[in]  mask     Macro name of the bit. See example for details.
+ * @param[in]  value     Data which is used to set the bits
+ *
+ * @return  updated \a reg_data
+ *
+ * @par  Example
+ * The following example uses (internally) the definitions \c TMAG5273_INT_STATE_MSK
+ * and \c TMAG5273_INT_STATE_POS to set the bits of the register \c INT_CONFIG_1.
+ *    @code{.c}
+ *       uint8_t  reg_int_config_1 = 0x03; // initialized to some value
+ *       reg_int_config_1 = TMAG5273_FIELD_SET(reg_int_config_1,
+ *           TMAG5273_INT_STATE, TMAG5273_INT_STATE_PULSE);
+ *    @endcode
+ */
+#define TMAG5273_FIELD_SET(reg_data, mask, value) \
+	((FIELD_PREP(mask, value)) | (reg_data & ~(mask)))
+
+/* -------------------------------------------------------------------- */
+
+/* --- structs --- */
+
+/** static configuration data */
+struct tmag5273_config {
+	struct i2c_dt_spec i2c;
+
+	bool x_axis;
+	bool y_axis;
+	bool z_axis;
+	bool temperature;
+
+	uint8_t meas_range;
+	uint8_t temperature_coefficient;
+	uint8_t angle_magnitude_axis;
+	uint8_t ch_mag_gain_correction;
+
+	uint8_t operation_mode;
+	uint8_t averaging;
+
+	bool trigger_conv_via_int;
+	bool crc_enabled;
+	bool low_power_mode;
+
+	struct gpio_dt_spec int_gpio;
+	uint8_t mask_intb_value;
+
+	bool need_int_pin;
+};
+
+/** dynamic data (stored in RAM) */
+struct tmag5273_data {
+	uint8_t version; /** version as given by the sensor */
+
+	int16_t x_sample; /** measured B-field @x-axis */
+	int16_t y_sample; /** measured B-field @y-axis */
+	int16_t z_sample; /** measured B-field @z-axis */
+	int16_t temperature_sample; /** measured temperature data */
+
+	uint16_t xyz_range; /** magnetic range for x/y/z-axis in mT */
+
+	int16_t angle_sample; /** measured angle in degree, if activated */
+	uint8_t magnitude_sample; /** Positive vector magnitude (can be >7 bit). */
+};
```

Comment:
You are right, but this is the first part of the driver. I have also the PM, the trigger handling as well as multiple TMAG5273 on one bus.

Since I stripped everything, keeping this in the header was the easiest way. If this isn't okay I will change it?